Enyo 2562: Common-ify Ares/Sdk generator in ares-generator.git

sparkleholic opened the pull request:
- CLI can have multiple options for generating like (overwrite, existed status, ...)
- If generate function is changed, it will affect genZip of ares-project.
  So, instead setOptions function is added.
- After generating have done, options will be reset.

Enyo-DCO-1.1-Signed-off-by: Junil Kim logyourself@gmail.com

sparkleholic committed:
ENYO-2562: added setOptions func for supporting CLI generator options

Enyo-DCO-1.1-Signed-off-by: Junil Kim <logyourself@gmail.com>

sparkleholic committed:
ENYO-2562: changed _resetOptions func location

Enyo-DCO-1.1-Signed-off-by: Junil Kim <logyourself@gmail.com>

asnowfix commented on ares-generator.js:
```diff
@@ -187,6 +192,29 @@ var shell = require("shelljs"),
+		},
+
+		setOptions: function(opt, next) {
+			if (typeof opt != 'object') {
```
Use `!==` instead of `!=`

asnowfix commented:
1. I prefer to add a new `options` parameter to `generate()` rather than a second method, because there is nothing to indicate that these options are specific to a single request. I will adapt `ares-project` accordingly when we switch to a modified ares-generator
2. Although I understand the role of the `overwrite:` option (should be true by default in the library, when undefined, to preserve the original behaviour), I do not understand the role of the `existed` option. Please clarify.
3. In the original version, I took care to avoid any `fs.xxxSync()` operation by using `async.waterfall()` and other `async.series()`. Could it be possible to stick with the same approach?

sparkleholic committed:
ENYO-2562: removed existed option

added options parameter into generate function

Enyo-DCO-1.1-Signed-off-by: Junil Kim <logyourself@gmail.com>

sparkleholic commented:
1. added `options` parameter to `generate()`
2. removed `existed` option.
3. It remains one `fs.xxxSync` function. I'll improve it later.

sparkleholic committed:
ENYO-2562: removed unnecessary local variable

Enyo-DCO-1.1-Signed-off-by: Junil Kim <logyourself@gmail.com>

sparkleholic committed:
ENYO-2562: fixed indent

Enyo-DCO-1.1-Signed-off-by: Junil Kim <logyourself@gmail.com>

sparkleholic committed:
ENYO-2562: generate works like overwriting as default when options.overwrite is not defined

Enyo-DCO-1.1-Signed-off-by: Junil Kim <logyourself@gmail.com>

sparkleholic commented:
@snowfix Hello, FiX. I'll be appreciate if you review this.
And it needs modifying getZip  when it call `ares-generator.generate()` , I think when creating project it can have `{overwrite:true}` or `{}` parameter.
And after merging this PR, It also needs to review&merge `https://github.com/enyojs/ares-webos-sdk/pull/55` for CLI.

Commit:
ENYO-2562: enable jshint & fix warnings

Enyo-DCO-1.1-Signed-off-by: Francois-Xavier KOWALSKI <francois-xavier.kowalski@hp.com>

asnowfix commented:
Merged. Will do the necessary changes into `ares-project`.
